File: app/src/main/java/com/example/ama/questapp/presentation/base/BasePresenter.java
package com.example.ama.questapp.presentation.base;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BasePresenter<T extends BaseContract.QuestView> implements BaseContract.Presenter<T> {
    protected T view;
    protected CompositeDisposable disposable;

    @Override
    public void attachView(T view) {
        this.view = view;
        disposable = new CompositeDisposable();
    }

    @Override
    public void detachView() {
        view = null;
        if (disposable != null) {
            disposable.dispose();
            disposable = null;
        }
    }
}

File: app/src/main/java/com/example/ama/questapp/presentation/base/BasePresenter.kt
package com.example.ama.questapp.presentation.base

import io.reactivex.disposables.CompositeDisposable
import moxy.MvpPresenter
import moxy.MvpView

abstract class BasePresenter<T : MvpView> : MvpPresenter<T>() {

    protected val compositeDisposable = CompositeDisposable()

    override fun onDestroy() {
        compositeDisposable.dispose()
        super.onDestroy()
    }
}